### ISSUE TITLE
What is a good way for hyperparameter tunings?

### ISSUE THREAD

**Issue by tedapham** [scikit-learn-contrib/hdbscan#316]
Hi, I'm trying to cluster about 2000 lat,long locations using driving distance. When used hdbscan in default mode, the number of cluster is usually 37-41 which are too many for my application. I wanted to get around 5-11 clusters so I ran 2 for loops for min_cluster_size, and min_samples from 1 to 60 and if the resulting number of cluster is within 5-11 then I save the results and compute the silhouette_score for the non-noise clusters. Then I selected the best combination of min_cluster_size and min_samples that gave me the largest silhouette_score. However, visually, the clusters don't often make sense
My question is, is there a better way to tune the parameters?
Thanks

**Comment by lmcinnes** [comment]
I think one option is to simply look at the full condensed tree that
results and work from there -- that gives you the best chance of getting
the granularity you want. I think playing with parameters and relying on
scores like silhouette can often lead to rather strange results.

**Comment by tedapham** [comment]
Thanks

I'm trying to get the relative_validity_ score but the object threw me an error, trying to set 'gen_min_span_tree = True' with metric=precomputed failed at assert len(candidates) > 0
